src/libs/actions/IOU.js

```diff
@@ -1816,6 +1827,26 @@ function setMoneyRequestMerchant(merchant) {
     Onyx.merge(ONYXKEYS.IOU, {merchant: merchant.trim()});
 }
 
+/**
+ * @param {Object} category
+ * @param {String} category.name
+ * @param {Boolean} category.enabled
+ * @param {String} policyID
+ * @param {Object[]} recentlyUsedPolicyCategories
+ * @param {String} recentlyUsedPolicyCategories[].name
+ * @param {Boolean} recentlyUsedPolicyCategories[].enabled
+ */
+function setMoneyRequestCategory(category, policyID, recentlyUsedPolicyCategories) {
+    const uniqueRecentlyUsedPolicyCategories = _.filter(recentlyUsedPolicyCategories, (recentlyUsedPolicyCategory) => recentlyUsedPolicyCategory.name !== category.name);
+
+    Onyx.merge(ONYXKEYS.IOU, {category: category.name});
+    Onyx.merge(`${ONYXKEYS.RECENTLY_USED_POLICY_CATEGORIES}${policyID}`, [category, ...uniqueRecentlyUsedPolicyCategories]);
```

Comment:
We are filtering `recentlyUsedPolicyCategories` to not have the category. But here we are merging both again. Instead why can't we just do merge only if the `recentlyUsedPolicyCategories` doesn't have the category?

Reply:
Hmmm... Not sure we want to do this here before saving the category. We only want to add the category to the list of recently used categories upon saving the expense. In other words, just because I have selected the category does not mean I have "used" it. I think this should go in the optimistic data for saving money request.